### ISSUE TITLE
Feature: Add Preview for Office files - Word, Excel

### ISSUE THREAD

**Issue by Timeless4k** [files-community/Files#4041]
Currently, you can only preview PDF file can we improve the preview menu to incorporate Word, Excel, TXT etc. When your click on the file you can see the preview of the file, just have the ability like the normal file explorer.

**Comment by gave92** [comment]
Hello, preview for .TXT files it's already implemented. If you don't see it it's a bug.

**Comment by winston-de** [comment]
I've been looking into this, and unfortunately it would be very difficult to get a preview that is on par with Windows Explorer, as it essentially embeds the Word (or other Office) application inside of the preview pane. And, unlike PDFs, there is no built in API to render them, and all the third party ones I've found are paid.
However, since there does seem to be a lot of interest in this, I will do my best to put something together.

**Comment by gave92** [comment]
There's a free version of Spire.Office but I don't think it's free for uwp :/
Might try to do it from the fulltrustprocess and send the render as images to uwp? Yuk

**Comment by generalguy41** [comment]
Not sure if it's been looked into already but Quicklook has an Office plugin. Maybe something from there can be used?

https://github.com/QL-Win/QuickLook.Plugin.OfficeViewer

**Comment by gave92** [comment]
Not directly as that's for wpf. But I believe we could try to get the equivalent for uwp: https://www.syncfusion.com/word-framework/uwp. License for open source projects might be free.

**Comment by winston-de** [comment]
@gave92 That looks perfect. 
It looks like the library has render as image capabilities for PowerPoint, and Excel files, so we could load and display those previews the same way as the PDF preview. 
Apparently Word document to image isn't support on UWP, so we would have to either show the document editor control (a bit overkill for a preview), or convert it to a PDF stream and render that instead (could be slower).

However, this all would require:
A). Acquiring a open source license (and possibly having to follow the terms of said license)
B). Taking on several new rather large dependencies

@yair100 What do you think?

**Comment by yaichenbaum** [comment]
@winston-de Is it possible to use extensions for this or will we still need to add the dependencies?

**Comment by winston-de** [comment]
@yair100 Extensions would work, as could an optional package.

**Comment by yaichenbaum** [comment]
@winston-de Let's do this via an extension in order to avoid bundling an extra dependency to Files.

**Comment by pli7pli7** [comment]
Hi all, 
Thanks for looking into this last year. Didn't realize the complexity behind the scenes. I thought it was just a setting issue on my PC.

Any progress or updates since a year ago? This (Word/PPT/Excel) is a much needed feature to be supported. Thanks.

**Comment by yaichenbaum** [comment]
@pli7pli7 we're currently focusing on performance improvements but we're still planning to add support for previewing Office files.

**Comment by pli7pli7** [comment]
Thanks @yair100 ~ any estimated timeline? possibly this year?

(asking because this is the only major missing feature before I could use Files as a full substitute for the built-in File Explorer)

**Comment by yaichenbaum** [comment]
> Thanks @yaichenbaum ~ any estimated timeline? possibly this year?
> 
> (asking because this is the only major missing feature before I could use Files as a full substitute for the built-in File Explorer)

@pli7pli7 we don't have a timeline since we need a contributor to work on it.

**Comment by Josh-65** [comment]
Is this an issue in the preview version? https://files.community/download/preview

**Comment by yaichenbaum** [comment]
Yes

**Comment by chezfo** [comment]
I saw that there is a "Everything" program that uses Office Preview

![image](https://user-images.githubusercontent.com/85172877/221525330-908e9bb5-6078-4d1e-9519-61f594520c06.png)

**Comment by yaira2** [comment]
Everything is probably using the system preview handler, we have quite a bit of work to do before we can do the same.

**Comment by yaira2** [comment]
This feature is implemented in `main` and will be included in the next release.

**Comment by EvoLandEco** [comment]
Now if we want to preview a xlsx file, we click on "View raw" and will be redirected to view.officeapps.live.com. 

However, I found that this web application will always load an earlier version of the same file regardless of whether I pushed an update to the xlsx. I tried to clean up browser cache and cookies, but it doesn't work, I re-open the repository, locate that xlsx file, click on "View raw" and I always open the same earlier version of the file.

I was wondering if this could be a bug, or just something cache related goes wrong on my end.
